src/Frontend/Modules/MailMotor/Actions/Subscribe.php

```diff
@@ -0,0 +1,252 @@
+use Frontend\Modules\MailMotor\Engine\Model as FrontendMailMotorModel;
+use MailMotor\Bundle\MailMotorBundle\Exception\NotImplementedException;
+
+/**
+ * This is the Subscribe-action for our MailMotor
+ *
+ * @author Jeroen Desloovere <jeroen@siesqo.be>
+ */
+class Subscribe extends FrontendBaseBlock
+{
+    /**
+     * FrontendForm instance
+     *
+     * @var	FrontendForm
+     */
+    private $frm;
+
+    /**
+     * @var array
+     */
+    private $interests;
+
+    /**
+     * Execute the extra
+     *
+     * @return void
+     */
+    public function execute()
+    {
+        parent::execute();
+        $this->loadData();
+        $this->loadTemplate();
+        $this->loadForm();
+        $this->validateForm();
+        $this->parse();
+    }
+
+    /**
+     * Load the form
+     *
+     * @return void
+     */
+    private function loadData()
+    {
+        try {
+            // Define interests
+            $this->interests = $this->get('mailmotor.subscriber')->getInterests();
+        // Fallback for when no mail-engine is chosen in the Backend
+        } catch (NotImplementedException $e) {
+            $this->interests = false;
+        }
+    }
+
+    /**
+     * Load the form
+     *
+     * @return void
+     */
+    private function loadForm()
+    {
+        // create the form
+        $this->frm = new FrontendForm('mailMotorSubscribeForm');
+
+        // define email
+        $email = null;
+
+        // request contains an email
+        if ($this->get('request')->request->get('email') != null) {
+            $email = $this->get('request')->request->get('email');
+        }
+
+        // create & add elements
+        $this->frm->addText('email', $email);
+        $this->frm->addText('email')
+            ->setAttributes(
+                array(
+                    'required' => null,
+                    'type' => 'email',
+                    'placeholder' => ucfirst(Language::lbl('YourEmail')),
+                )
+            )
+        ;
+
+        // Has interests
+        if ($this->interests) {
+            // Init
+            $chkInterestValues = array();
+
+            // Loop interests
+            foreach ($this->interests as $categoryId => $categoryInterest) {
+                foreach ($categoryInterest['children'] as $categoryChildId => $categoryChildTitle) {
+                    // Add interest value for checkbox
+                    $chkInterestValues[] = [
+                        'value' => $categoryChildId,
+                        'label' => $categoryChildTitle,
+                    ];
+                }
+            }
+
+            $this->frm->addMultiCheckbox(
+                'interests',
+                $chkInterestValues
+            );
+        }
+    }
+
+    /**
+     * Parse the data into the template
+     *
+     * @return void
+     */
+    protected function parse()
+    {
+        // form was sent?
+        if ($this->URL->getParameter('sent') == 'true') {
```

Comment:
using sent is a bad practice since it is too generic and may clash with other widgets on the page